org.opent2t.sample.hub.superpopular/com.nest.hub/js/thingTranslator.js

```diff
@@ -1,69 +1,111 @@
-/* jshint esversion: 6 */
-/* jshint node: true */
-/* jshint sub:true */
 // This code uses ES2015 syntax that requires at least Node.js v4.
 // For Node.js ES2015 support details, reference http://node.green/
 
 "use strict";
-var request = require('request-promise');
+var OpenT2T = require('opent2t').OpenT2T;
+var Firebase = require("firebase");
```

Comment:
We moved to firebase to lift restrictions like rate limits for REST streaming, please see [Nest dev page](https://developers.nest.com/documentation/cloud/rest-streaming-guide) for details. However, one thing to noted that Nest only works with an older version of firebase, as seen at [this Nest forum post](https://nestdevelopers.io/t/how-can-we-integrate-nest-using-firebase-3-5-javascript-sdk/325/2) .